tests/post/steps/test_volume.py

```diff
@@ -48,13 +48,57 @@ def test_deploy_operator(host):
 def test_volume_creation(host):
     pass
 
+@scenario('../features/volume.feature',
+          'Test volume deletion (sparseLoopDevice)')
+def test_volume_deletion(host):
+    pass
+
+# }}}
+# Given {{{
+
+@given(parsers.parse("a Volume '{name}' exists"))
+def volume_exist(host, name, k8s_custom_client):
```

Comment:
nested function are usually defined at the top of the outer function, not in the middle:

```python
def volume_exist(host, name, k8s_custom_client):
    def _check_availability():
        volume = _get_volume(k8s_custom_client, name)
        assert volume['status']['phase'] == 'Available',\
            'Volume not available, status={}'.format(volume['status']['phase'])

    volume = _get_volume(k8s_custom_client, name)
    if volume is not None:
        return
    body = DEFAULT_VOLUME.format(name=name)
    _create_volume(k8s_custom_client, body)
    utils.retry(
        _check_availability, times=30, wait=2,
        name='checking availability of Volume {}'.format(name)
    )
```